crates/api/api_crud/src/user/create.rs

```diff
@@ -615,3 +654,110 @@ fn check_code_verifier(code_verifier: &str) -> LemmyResult<()> {
     Err(LemmyErrorType::InvalidCodeVerifier.into())
   }
 }
+
+/// Fetch list of communities from lemmyverse.net and then fetch the top 1000 communities.
+fn fetch_community_list(context: Data<LemmyContext>) {
+  spawn_try_task(async move {
+    let communities_json = include_str!(concat!(env!("OUT_DIR"), "/communities.json"));
+    let communities: Vec<ObjectId<ApubCommunity>> = serde_json::from_str(communities_json)?;
+
+    // Fetch communities themselves
+    let tasks = communities.iter().map(|c| async {
+      let context = context.reset_request_count();
+      c.dereference(&context).await.ok();
+    });
```

Comment:
Is this actually HTTP fetching data from maybe hundreds of communities? 

Really we should only be creating instance and community rows locally, as this issue is about community discovery. IE instead of hammering these communities with fetches, we should be running `Instance::read_or_create` and `Community::create` with the supplied data.

Reply:
What's the purpose of those http fetches? Don't we only need to fill the community table rows so that the communities are searchable? In that case we only need to do DB inserts.

Reply:
Alright then, I spose its okay.